shell/browser/api/electron_api_web_contents.cc

```diff
@@ -2155,8 +2155,10 @@ void WebContents::DevToolsOpened() {
   // Inherit owner window in devtools when it doesn't have one.
   auto* devtools = inspectable_web_contents_->GetDevToolsWebContents();
   bool has_window = devtools->GetUserData(NativeWindowRelay::UserDataKey());
-  if (owner_window() && !has_window)
-    handle->SetOwnerWindow(devtools, owner_window());
+  if (!has_window) {
+    if (const auto* window = owner_window())
```

Comment:
I don't understand how this fixes a case where `owner_window()` returns a dangling pointer?

The `non-nullptr` test on line 2159 will succeed, so both the old and new version of this function will pass the invalid pointer on to `SetOwnerWindow()`